grunt/config/replace.js

```diff
@@ -1,19 +1,46 @@
+var path = require('path');
+var fs = require('fs');
+
 module.exports = function (grunt, options) {
 
-  var courseDir = options.sourcedir + 'course/';
+  var courseDir = path.join(options.sourcedir, 'course');
 
-  if (options.outputdir.indexOf('build/') > -1) {
-    courseDir = options.outputdir + 'course/';
+  if (options.outputdir.indexOf('build') > -1) {
+    courseDir = path.join(options.outputdir, 'course');
   }
 
-  var configJson = grunt.file.readJSON(courseDir + 'config.json');
-  var defaultLanguage = configJson._defaultLanguage || 'en';
-  var courseJson = grunt.file.readJSON(courseDir + defaultLanguage + '/course.json');
+  var pathToConfig = path.join(courseDir, 'config.json')
 
-  // Backwards compatibility for courses missing 'description'
-  if (!courseJson.hasOwnProperty('description')) {
-    courseJson.description = '';
+  try {
+    // Verify that the configuration file exists.
+    fs.accessSync(pathToConfig);
+  } catch(ex) {
+    // If it does not, early return an empty configuration object, 
+    // so the replace task can still run.
+    return {
+      dist: {
+        options: {
+          patterns: [
+            {
+              json: {}
+            }
+          ]
+        },
+        files: [
+          {
+            expand: true,
+            flatten: true,
+            src: [options.outputdir + '*.xml'],
+            dest: options.outputdir
+          }
+        ]
+      }
+    }
   }
+  
+  var configJson = grunt.file.readJSON(pathToConfig);
```

Comment:
Would it be better to wrap this line in a try catch and set configJson to an empty object on error? Is is possible to have a circumstance where a replacement task should still be executed with the data in courseJson even if configJson is missing?